packages/ui/src/components/Form/customField/PasswordField.tsx

```diff
@@ -0,0 +1,214 @@
+        !searchInputRef.current?.contains(event.target as Node)
+      ) {
+        setIsAutocompleteOpen(false);
+      }
+      if (
+        !isAutocompleteOpen &&
+        searchInputRef &&
+        searchInputRef.current &&
+        searchInputRef.current.contains(event.target as Node)
+      ) {
+        setIsAutocompleteOpen(true);
+      }
+    },
+    [isAutocompleteOpen],
+  );
+
+  useEffect(() => {
+    window.addEventListener('keydown', handleMenuKeys);
+    window.addEventListener('click', handleClickOutside);
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [handleClickOutside, handleMenuKeys, isAutocompleteOpen]);
+
+  const textInput = (
+    <div ref={searchInputRef} id="password-input">
+      <InputGroup>
+        <InputGroupItem isFill>
+          <TextInput
+            aria-label={'uniforms text field'}
+            data-testid={'password-field'}
+            name={props.name}
+            isDisabled={props.disabled}
+            validated={props.error ? 'error' : 'default'}
+            onChange={handleOnChange}
+            placeholder={props.placeholder}
+            ref={props.inputRef}
+            type={passwordHidden ? 'password' : 'text'}
+            value={props.value}
+            {...filterDOMProps(props)}
+          />
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem
+                icon={props.error && <ExclamationCircleIcon />}
+                variant={props.error ? 'error' : 'default'}
+              >
+                {!props.error ? props.helperText : props.errorMessage}
+              </HelperTextItem>
+            </HelperText>
+          </FormHelperText>
+        </InputGroupItem>
+        <InputGroupItem>
+          <Button
+            data-testid={'password-show-hide-button'}
+            variant="control"
+            onClick={() => setPasswordHidden(!passwordHidden)}
+            aria-label={passwordHidden ? 'Show' : 'Hide'}
+          >
+            {passwordHidden ? <EyeIcon /> : <EyeSlashIcon />}
+          </Button>
+        </InputGroupItem>
+      </InputGroup>
+    </div>
+  );
+
+  const autocomplete = (
+    <Menu ref={autocompleteRef} onSelect={onSelect}>
+      <MenuContent>
+        <MenuList>
+          <MenuItem
+            data-testid={'password-use-suggested-menuitem'}
+            itemId={0}
+            actions={
+              <MenuItemAction
+                icon={<RedoIcon aria-hidden />}
+                onClick={(_e) => {
+                  setGeneratedPassword(generatePassword());
+                }}
+                actionId="redo"
+                aria-label="Generate a new suggested password"
+              />
+            }
+          >
+            Use suggested password: <b>{`${generatedPassword}`}</b>
+          </MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
```

Comment:
Considering that we need to provide existing passwords and that we're not creating new user accounts through Kaoto (_at least not at the moment_) I wonder if we would be better off without the password generator bit.

Reply:
TBH I have no idea from what background we wanted to introduce password generator, @lhein could you add some context?

Reply:
The idea was that this component provides the visibility switch and the masking functionality ootb. If you can come up with something similar I am not opposed to hear your solution :)

Reply:
OK so no need to use PF password generator at all, visibility toggle is just a plain button and the masking is HTML standard. I'll remove the password generator.